test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDropDownListExtensionsTest.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.TestCommon;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Core
+{
+    public class HtmlHelperDropDownListExtensionsTest
+    {
+        private static readonly List<SelectListItem> BasicSelectList = new List<SelectListItem>
+        {
+            new SelectListItem { Text = "Zero",  Value = "0"},
+            new SelectListItem { Text = "One",   Value = "1"},
+            new SelectListItem { Text = "Two",   Value = "2"},
+            new SelectListItem { Text = "Three", Value = "3"},
+        };
+
+        [Fact]
+        public void DropDownList_FindsSelectList()
+        {
+            // Arrange
+            var expectedHtml = "<select id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\">" +
+                "<option value=\"HtmlEncode[[0]]\">HtmlEncode[[Zero]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[1]]\">HtmlEncode[[One]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[2]]\">HtmlEncode[[Two]]</option>" + Environment.NewLine +
+                "<option value=\"HtmlEncode[[3]]\">HtmlEncode[[Three]]</option>" + Environment.NewLine +
+                "</select>";
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property1 = BasicSelectList };
```

Comment:
Why does `TestModel` have a property of type `IEnumerable<SelectListItem>`? If it were a view model, the property would _never_ be looked up in this manner; the view model property would be used explicitly.

The usual patterns are
``` c#
class ViewModel
{
    string Property1 { get; set; }
    IEnumerable<SelectListItem> Property1SelectList { get; }
}
...
@Html.DropDownListFor(m => m.Property1, Model.Property1SelectList)
```
or
``` c#
class RealModel
{
    string Property1 { get; set; }
}
...
public IActionResult Index()
{
    ViewData["Property1"] = new SelectList(...);
    return View();
}
...
@Html.DropDownListFor(m => m.Property1)
```